modules/reindex/src/test/java/org/opensearch/index/reindex/ReindexRenamingSettingTests.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.reindex;
+
+import org.junit.Before;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A unit test to validate the former name of the setting 'reindex.remote.allowlist' still take effect,
+ * after it is deprecated, so that the backwards compatibility is maintained.
+ * The test can be removed along with removing support of the deprecated setting.
+ */
+public class ReindexRenamingSettingTests extends OpenSearchTestCase {
+    ReindexPlugin plugin;
+
+    @Before
+    public void setup() {
+        this.plugin = new ReindexPlugin();
+    }
+
+    public void testReindexSettingsExist() {
+        List<Setting<?>> settings = plugin.getSettings();
+        assertTrue(
+            "Both 'reindex.remote.allowlist' and its predecessor should exist in settings of Reindex plugin",
+            settings.containsAll(
+                Arrays.asList(TransportReindexAction.REMOTE_CLUSTER_WHITELIST, TransportReindexAction.REMOTE_CLUSTER_ALLOWLIST)
+            )
+        );
+    }
+
+    public void testSettingFallback() {
+        assertEquals(
+            TransportReindexAction.REMOTE_CLUSTER_ALLOWLIST.get(Settings.EMPTY),
+            TransportReindexAction.REMOTE_CLUSTER_WHITELIST.get(Settings.EMPTY)
+        );
+    }
+
+    public void testSettingGetValue() {
+        Settings settings = Settings.builder().put("reindex.remote.allowlist", "127.0.0.1:*").build();
+        assertEquals(TransportReindexAction.REMOTE_CLUSTER_ALLOWLIST.get(settings), Arrays.asList("127.0.0.1:*"));
+        assertEquals(TransportReindexAction.REMOTE_CLUSTER_WHITELIST.get(settings), Arrays.asList());
+    }
+
+    public void testSettingGetValueWithFallback() {
+        Settings settings = Settings.builder().put("reindex.remote.whitelist", "127.0.0.1:*").build();
```

Comment:
What happens if both "reindex.remote.allowlist" and "reindex.remote.whitelist" are specified but they define different values?

Reply:
Hi @andrross , thanks for your review! I have got the concern here, I know it will make the user confused when using the 2 different expression for this setting. The confusion is little when the setting is [static](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/settings.html#cluster-setting-types) - can only be configured through `opensearch.yml` file, but it will be larger when the setting is dynamic - can be configured in both "cluster setting API" and `opensearch.yml` file.
All usages of the "whitelist" setting in the code base have been changed to the "allowlist" setting, so I think the value of the "whitelist" will be ignored, once "allowlist" is configured.

Reply:
The unit test here also shows the behavior,
https://github.com/opensearch-project/job-scheduler/pull/20/files#diff-2d466c782fd61b8154ef2e4ec067b3d528cc92c9ce6cd4d6401475568077abf1R87
Although they can have different value, with my code change, value of the old "whitelist" setting will not be taken anywhere.

Reply:
This was actually a bit of a leading question :) Can you create a unit test to demonstrate/validate this behavior? For example, if you set both settings to different values and use `TransportReindexAction.REMOTE_CLUSTER_ALLOWLIST`, only the "allowlist" value should be returned because the fallback setting should not be used.

Relatedly, can you make the `TransportReindexAction.REMOTE_CLUSTER_WHITELIST` constant private? I believe it should not be directly used anywhere and the only access to that setting is via the fallback mechanism of `TransportReindexAction.REMOTE_CLUSTER_ALLOWLIST`. Please correct me if I'm wrong.

Reply:
Thanks for your suggestion! I probably can just modify the test case above: `testSettingGetValue()` to give a different value of the "whitelist" setting, and then get both of the setting value by calling `TransportReindexAction.REMOTE_CLUSTER_*LIST`.

I just noticed there are deprecated setting be set as private variable, such as https://github.com/opensearch-project/OpenSearch/blob/1.2.4/server/src/main/java/org/opensearch/node/Node.java#L229, this is a good idea to me! I will learn more impact about it.
Currently the "whitelist" setting is still added into the Settings list of the plugin (https://github.com/opensearch-project/OpenSearch/blob/c8dfd682d000cc7716302615f6ebd4e3cf4bb337/modules/reindex/src/main/java/org/opensearch/index/reindex/ReindexPlugin.java#L135), I'm trying to understand the exact reason.

Reply:
> Currently the "whitelist" setting is still added into the Settings list of the plugin

You probably don't need that since "whitelist" is configured as a fallback for the allowlist option, right?

Reply:
I'm worried that if removing the Line 135 `settings.add(TransportReindexAction.REMOTE_CLUSTER_WHITELIST); `, any custom value of the "whitelist" setting will be ignored, which leads "allowlist" setting fallback to the default value of "whitelist" setting. Let me validate through the unit test.

Reply:
I don't understand. The desired behavior is that if 'allowlist' is specified then 'whitelist' should be ignored. If only 'whitelist' is specified then it will be used. Do we still get that behavior if your line 135?

Reply:
Hi @andrross, I will do an experiment in a real cluster to validate whether the Line 135 is necessary or not. I'm only afraid that without putting that setting in the "known settings" list, there will be an error on start up. (https://www.elastic.co/guide/en/elasticsearch/reference/7.15/archived-settings.html)

Reply:
Yes, please do that quick experiment because it would make the code much cleaner (and the tests simpler) to just have that fallback setting as a private field.

Reply:
I assembled an linux tar distribution of OpenSearch, and confiugred `reindex.remote.whitelist` in `opensearch.yml`, then start the node, but the error appears.
But I can restrict the use of constant `TransportReindexAction.REMOTE_CLUSTER_WHITELIST` in the same package.
```
[2022-02-24T23:10:08,579][ERROR][o.o.b.OpenSearchUncaughtExceptionHandler] [ip-172-31-15-168] uncaught exception in thread [main]
org.opensearch.bootstrap.StartupException: java.lang.IllegalArgumentException: unknown setting [reindex.remote.whitelist] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
	at org.opensearch.bootstrap.OpenSearch.init(OpenSearch.java:182) ~[opensearch-2.0.0-SNAPSHOT.jar:2.0.0-SNAPSHOT]
	at org.opensearch.bootstrap.OpenSearch.execute(OpenSearch.java:169) ~[opensearch-2.0.0-SNAPSHOT.jar:2.0.0-SNAPSHOT]
...
uncaught exception in thread [main]
java.lang.IllegalArgumentException: unknown setting [reindex.remote.whitelist] please check that any required plugins are installed, or check the breaking changes documentation for removed settings
	at org.opensearch.common.settings.AbstractScopedSettings.validate(AbstractScopedSettings.java:589)
...
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:135)
	at org.opensearch.bootstrap.OpenSearch.main(OpenSearch.java:101)
```